docs/developers/reproducible-contract-builds.md

```diff
@@ -0,0 +1,203 @@
+:::
+
+### Choose an image tag
+
+The first step for supporting reproducible builds is to decide on a specific Docker image tag to use. Check the **frozen** tags listed at [elrondnetwork/elrond-sdk-erdpy-rust](https://hub.docker.com/r/elrondnetwork/elrond-sdk-erdpy-rust/tags), and inspect their labels - especially the label `rust`:
+
+```
+LABEL rust=nightly-2022-03-01
+```
+
+The chosen, _frozen_ image tag should accompany the versioned source code (e.g. via _release notes_), in order to inform others on how to reproduce a specific build (of a specific source code version). In this context, _frozen_ image tag refers to a Docker image tag that will never get any updates after its initial publishing.
+
+:::tip
+If unsure about which one to choose - pick the tag with the **largest index number**. 
+:::
+
+:::warning
+Never pick the tag called `latest` for production-ready builds.
+:::
+
+### Regular builds vs. _deployment_ build
+
+Usually, while developing, testing and integrating a smart contract, one would simply run locally the command `erdpy contract build`, which generates the WASM bytecode. 
+
+However, for production-ready deployments, in order to achieve reproducible builds, make sure to only deploy contracts built as described in the next section, using the chosen Docker image (tag).
+
+## (Re)producing a build
+
+In this section, you'll learn how to run a reproducible build, or, to put it differently, how to reproduce a previous build (made by you or by someone else in the past), on the local machine.
+
+### Fetch the source code
+
+Let's clone the source code locally, and switch to a certain version that we'd like to build:
+
+```
+mkdir -p ~/contracts && cd ~/contracts
+git clone https://github.com/ElrondNetwork/reproducible-contract-build-example.git --branch=v0.1.2 --depth=1
+```
+
+### Preparing environment variables
+
+Let's export the following variables:
+
+```
+export REPOSITORY_ON_HOST=~/contracts/reproducible-contract-build-example
+export REPOSITORY_IN_CONTAINER=/home/elrond/reproducible-contract-build-example
+export IMAGE=elrondnetwork/elrond-sdk-erdpy-rust:frozen-003
+```
+
+The latter export statement explicitly selects the **chosen, _frozen_ Docker image tag** to be used.
+
+:::important
+As of September 2022, you should stick to using the prefix `/home/elrond` for `REPOSITORY_IN_CONTAINER`.
+:::
+
+### Performing the build
+
+Make sure the folder `REPOSITORY_ON_HOST` is accessible from Docker. You can easily do so by granting read & write permissions for all users:
+
+```
+sudo chmod -R a+rw ${REPOSITORY_ON_HOST}
+```
+
+Now let's build the contract by invoking the following `docker run` command:
+
+```
+docker run -it \
```

Comment:
Can we have a TLDR with the steps of simply generating e deterministic build? So people who don't care about the reasoning can just copy/paste some commands and build the contract.

As a side note, in `create-erd-app` (after today's changes simply `erd-app`) I added a docker-compose file, so generating a deterministic build (for devs using this tool) will be as simple as `npx erd-app contract build`

Reply:
Nice :tada: 

Fixed.